src/main/java/com/verily/genomewarp/utils/GvcfToVcfAndBed.java

```diff
@@ -19,92 +19,145 @@
 import java.io.IOException;
 import java.io.PrintWriter;
 import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
- * Helper class for extracting a canonical VCF and a BED file from a gVCF file
+ * Helper class for extracting a canonical VCF and a BED file from a gVCF file.
  */
 public class GvcfToVcfAndBed {
-
-  private static final Logger logger = Logger.getLogger(VcfToVariant.class.getName());
-
+  // class constants
+  private static final Logger logger = Logger.getLogger(GvcfToVcfAndBed.class.getName());
+  
+  private static final int MIN_VCF_FIELDS = 8;
+  private static final int START_FIELD_INDEX = 1;
+  private static final int END_FIELD_INDEX = 3;
+  private static final int ALT_BASES_FIELD_INDEX = 4;
+  private static final int FILTER_FIELD_INDEX = 6;
+  private static final int FORMAT_FIELD_INDEX = 7;
+  private static final String UNKNOWN_VALUE = ".";
+  private static final String GVCF_ALT_ALLELE = "<*>";
+  private static final String HEADER_START_STR = "#";
+  private static final String FIELD_SEP_REGEX = "\\t";
+  private static final String PASS_FILTER_VALUE = "PASS";
+  private static final String FORMAT_FIELD_SEP_REGEX = ";";
+  private static final String FORMAT_END_KEY = "END";
+  private static final int CHROM_FIELD_INDEX = 0;
+  private static final String BED_FIELD_SEP_STR = "\t";
+  
   /**
    * Solve Issue #2 - Extract variant VCF and BED from (tentative) gVCF file
    * 
-   * @param rawQueryVcf user-provided (g?)VCF path
+   * @param rawQueryGvcf user-provided gVCF path
    * @param vcfFromGvcf intermediate canonical VCF path
    * @param bedFromGvcf intermediate BED path
-   * @return true if query VCF is a gVCF, false otherwise
+   * @return true if query gVCF was parsed successfully, false otherwise
    */
-  public static boolean saveVcfAndBedFromGvcf(String rawQueryVcf, String vcfFromGvcf,
-      String bedFromGvcf) throws IOException {
-    PrintWriter outBed = new PrintWriter(Files.newBufferedWriter(Paths.get(bedFromGvcf), UTF_8));
-    PrintWriter outVcf = new PrintWriter(Files.newBufferedWriter(Paths.get(vcfFromGvcf), UTF_8));
-    BufferedReader br = Files.newBufferedReader(Paths.get(rawQueryVcf), UTF_8);
+  public static boolean saveVcfAndBedFromGvcf(String rawQueryGvcf, String vcfFromGvcf,
+      String bedFromGvcf) {
     String line;
-    boolean isGvcf = false;
+    boolean isSuccessfulGvcf = false;
```

Comment:
How about

boolean successfullyParsedGvcf = true;

And then only set the boolean to false within catch blocks?

Reply:
I can refactor the name but I need it to be true only when I have at least one BED line.

Reply:
Why? If a gVCF only contains non-passing variants the expected result is a VCF with the variant calls and an empty BED file. Looking through the downstream code I believe it handles an existing but empty BED file appropriately (nothing gets transformed).

Reply:
OK then, I was in doubt whether an empty BED file was a valid use case.